client/src/components/SignUp/index.tsx

```diff
@@ -15,22 +18,34 @@ const SignUp = () => {
   const [password, setPassword] = useState('');
   const [reenteredPassword, setReenteredPassword] = useState('');
 
+  const [duplicateKeys, setDuplicateKeys] = useState<Partial<FieldValues>>({});
+  const errors = getDuplicateKeyErrors(duplicateKeys, { email, studentNumber });
   const [isInfoEntered, setIsInfoEntered] = useState(false);
   const [isSignupComplete, setIsSignupComplete] = useState(false);
 
   const onSignUp = async () => {
     const watIAMUserId = email.split('@')[0];
-    await signup({
-      firstName,
-      lastName,
-      studentNumber: +studentNumber,
-      email,
-      watIAMUserId,
-      semester,
-      faculty,
-      password,
-    });
-    setIsSignupComplete(true);
+    try {
+      await signup({
+        firstName,
+        lastName,
+        studentNumber: +studentNumber,
+        email,
+        watIAMUserId,
+        semester,
+        faculty,
+        password,
+      });
+      setIsSignupComplete(true);
+    } catch (err) {
+      const error = err as RequestError<{
+        duplicateKeys?: Partial<FieldValues>;
+      }>;
+      if (error.data.duplicateKeys) {
+        setIsInfoEntered(false);
+        setDuplicateKeys(error.data.duplicateKeys);
+      }
```

Comment:
Right now I'm not rethrowing the error even if it does not contain the `duplicateKeys` field (maybe error'd for a different reason). So the request will fail but the client won't crash. I guess we really should throw if it's not an error we explicitly handle but we also need to set up Error boundaries so the whole app doesn't crash.

Reply:
Good thinking. This looks good
I think we could have an error boundary added in the near future. 
Will ticket